drivers/pds/targetcluster/targetcluster.go

```diff
@@ -47,35 +48,54 @@ type TargetCluster struct {
 	kubeconfig string
 }
 
-// RegisterToControlPlane register the target cluster to control plane.
-func (targetCluster *TargetCluster) RegisterToControlPlane(controlPlaneURL string, helmChartversion string, bearerToken string, tenantID string, clusterType string) error {
-	log.Info("Test control plane url connectivity.")
-	_, err := isReachbale(controlPlaneURL)
+//RegisterToControlPlane register the target cluster to control plane.
+func (targetCluster *TargetCluster) RegisterToControlPlane(controlPlaneURL string, helmChartversion string, bearerToken string, tenantId string, clusterType string) error {
+	var cmd string
+	apiEndpoint := fmt.Sprintf(controlPlaneURL + "api")
+	isRegistered := false
+	pods, err := k8sCore.GetPods(PDSNamespace, nil)
 	if err != nil {
-		return fmt.Errorf("unable to reach the control plane with following error - %v", err)
+		return err
 	}
-	var cmd string
-	apiEndpoint := fmt.Sprintf(controlPlaneURL + "/api")
-	log.Infof("Installing PDS ( helm version -  %v)", helmChartversion)
-	if strings.EqualFold(clusterType, "ocp") {
-		cmd = fmt.Sprintf("helm install --create-namespace --namespace=%s pds pds-target --repo=https://portworx.github.io/pds-charts --version=%s --set platform=ocp --set tenantId=%s "+
-			"--set bearerToken=%s --set apiEndpoint=%s --kubeconfig %s", PDSNamespace, helmChartversion, tenantID, bearerToken, apiEndpoint, targetCluster.kubeconfig)
-	} else {
-		cmd = fmt.Sprintf("helm install --create-namespace --namespace=%s pds pds-target --repo=https://portworx.github.io/pds-charts --version=%s --set tenantId=%s "+
-			"--set bearerToken=%s --set apiEndpoint=%s --kubeconfig %s", PDSNamespace, helmChartversion, tenantID, bearerToken, apiEndpoint, targetCluster.kubeconfig)
+	if len(pods.Items) > 0 {
+		log.InfoD("Target cluster is already registered to control plane.")
+		isRegistered = true
+	}
+	if !isRegistered {
+		log.Infof("Installing PDS ( helm version -  %v)", helmChartversion)
+		if strings.EqualFold(clusterType, "ocp") {
+			cmd = fmt.Sprintf("helm install --create-namespace --namespace=%s pds pds-target --repo=https://portworx.github.io/pds-charts --version=%s --set platform=ocp --set tenantId=%s "+
```

Comment:
since only --set platform=ocp is only difference you can have this logic
```
cmd = fmt.Sprintf("helm install --create-namespace --namespace=%s pds pds-target --repo=https://portworx.github.io/pds-charts --version=%s --set tenantId=%s "+
				"--set bearerToken=%s --set apiEndpoint=%s", PDSNamespace, helmChartversion, tenantId, bearerToken, apiEndpoint)

if strings.EqualFold(clusterType, "ocp") {

cmd = fmt.Sprintf("%s %s ",cmd,"--set platform=ocp")
}
```

Reply:
done